docs/wiki/PipelinesDesign.md

```diff
@@ -99,23 +110,27 @@ For additional details on how removal works please refer to the dedicated [Remov
 
 ### Publish
 
-The publish phase concludes each module's pipeline. If all previous tests succeeded (i.e. no phase failed) and the pipeline was executed in the [main\|master] branch, a new module version is published to all configured target locations. Currently we support
+The publish phase concludes each module's pipeline. If all previous tests succeeded (i.e. no phase failed) and the pipeline was executed in the `main` or `master` branch, a new module version is published to all configured target locations. Currently we support the following target locations:
+
 - _template specs_
 - _private bicep registry_
 - _universal artifacts_ (_Azure DevOps_ only)
 
-By the time of this writing, the publishing experience works as follows:
-1. A user can optionally specify a version in the module's pipeline file, or during runtime. If the user does not, a default version is used
-1. No matter what publishing location we enabled, the corresponding logic will
-   1. Fetch the latest version of this module in the target location (if available)
-   1. Compare it with any specified custom version the user optionally provided
-      - If the custom version is higher, it is used going forward
-      - If it is lower, the fallback mechanism will select a new version based on some default behavior (e.g. increment to the next patch version)
-   1. The identified new version is then used to publish the module to the target location in question
+Besides the execution of a publish, there is also the possibility to set the switch `Publish prerelease module`. This switch makes it possible to publish a pre-release version in every workflow run that is not based on main or master.
+
+> **Note**<br>
+> The `version` used for publishing any artifact will be the same in all three target locations which reduces the maintenance effort.
+
+The publishing works as follows:
+
+1. The script `utilities/pipelines/resourcePublish/Get-ModulesToUpdate.ps1` gets all changed module files including child modules and handles the logic of propagating the appropriate module version to be used:
+   1. The major (`x.0`) and minor (`0.x`) version are set based on the file `version.json` in the module folder.
+   1. The patch (`0.0.x`) version is calculated based on the number of commits on the `HEAD` ref. This Will cause the patch version to never reset to 0 with major and/or minor increment, as specified for [semver](https://semver.org/).
```

Comment:
```suggestion
   1. The patch (`0.0.x`) version is calculated based on the number of commits on the `HEAD` ref. This will cause the patch version to never reset to 0 with major and/or minor increment, as specified for [semver](https://semver.org/).
```